### PR TITLE
added symmetrize function

### PR DESCRIPTION
**Pull request by spinkney**
## Summary

Wrapped Eigen's selfadjointView function.

## Tests

1. Throw error if not a square matrix.
2. Check if result of symmetrize function is symmetric.

## Side Effects

None

## Release notes

Add `symmetrize(matrix m)` function to `stan::math`. 

## Checklist
- [x] Copyright holder: Sean Pinkney

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: stan/math/prim/fun/symmetrize.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Return a symmetric matrix using elements from the lower triangular part of
+ * the input matrix.
+ *
+ * @tparam T type of elements in the matrix
+ * @param m Matrix.
+ * @throw std:invalid_argument if the matrix is not square.
+ */
+template <typename T, require_eigen_t<T>* = nullptr>
+inline Eigen::Matrix<value_type_t<T>, Eigen::Dynamic, Eigen::Dynamic>
+symmetrize(const T& m) {
```
Comment (rok-cesnovar) [inline stan/math/prim/fun/symmetrize.hpp:20]: Should we maybe rename this to `symmetrize_low` or `symmetrize_to_upp` (not sure which one would be more descriptive) and then implement the upper versions as well (though that can be a separate PR).

**Review by rok-cesnovar — CHANGES_REQUESTED** [review]
Looks good but I have a suggestion to make the name a bit more descriptive. Not sure on what the best name would be.

File: stan/math/prim/fun/symmetrize.hpp (hunk 1 above)
Comment (spinkney) [reply stan/math/prim/fun/symmetrize.hpp:20]: I think you're right that it should be named differently, I just don't know a good non-verbose name for it. Maybe, `symmetrize_lower_tri`?

Comment (rok-cesnovar) [reply stan/math/prim/fun/symmetrize.hpp:20]: That seems like a good name to me. So `symmetrize_lower_tri()` would mean the lower triangular matrix gets copied to upper triangular.

Comment (rok-cesnovar) [reply stan/math/prim/fun/symmetrize.hpp:20]: @t4c1 suggested `symmetrize_from_lower_tri()`. What do you think?

Comment (spinkney) [reply stan/math/prim/fun/symmetrize.hpp:20]: Either one is good for me. I like the first one best for clarity.

Comment (rok-cesnovar) [reply stan/math/prim/fun/symmetrize.hpp:20]: Ok, lets go with that then.

File: stan/math/prim/fun.hpp [hunk 2]
```diff
@@ -305,6 +305,7 @@
 #include <stan/math/prim/fun/sub_row.hpp>
 #include <stan/math/prim/fun/subtract.hpp>
 #include <stan/math/prim/fun/sum.hpp>
+#include <stan/math/prim/fun/symmetrize.hpp>
```
Comment (rok-cesnovar) [inline stan/math/prim/fun.hpp:308]: The new file needs to be added here and the old one (symmetrize.hpp) removed from prim/fun.hpp as well as the files/tests deleted. Then this is good to go.

Comment (spinkney) [reply stan/math/prim/fun.hpp]: Yes, I'm on my mac and trying out github desktop. Typically I use tortoisegit on windows and when I brought commits over here it didn't update the files but added completely new ones! Anyway, should be fixed now.

**Review by rok-cesnovar — APPROVED** [review]
Thank you! Looks good. Congrats on your first successful Stan Math pull request! :tada:

File: stan/math/prim/fun.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/prim/fun.hpp]: No worries. Thanks.
